backend/pkg/kafka/config_helper.go

```diff
@@ -146,12 +147,37 @@ func NewKgoConfig(cfg *config.Kafka, logger *zap.Logger, hooks kgo.Hook) ([]kgo.
 
 		// AWS MSK IAM
 		if cfg.SASL.Mechanism == config.SASLMechanismAWSManagedStreamingIAM {
-			mechanism := aws.Auth{
-				AccessKey:    cfg.SASL.AWSMskIam.AccessKey,
-				SecretKey:    cfg.SASL.AWSMskIam.SecretKey,
-				SessionToken: cfg.SASL.AWSMskIam.SessionToken,
-				UserAgent:    cfg.SASL.AWSMskIam.UserAgent,
-			}.AsManagedStreamingIAMMechanism()
+			var mechanism sasl.Mechanism
+			if cfg.SASL.AWSMskIam.AccessKey != "" &&
```

Comment:
I'd say || rather than &&, or maybe even more clearly -- error if only one exists.

Reply:
Good point. Used the existing validate func to capture this requirement on config